log4j-api/src/main/java/org/apache/logging/log4j/status/StatusLogger.java

```diff
@@ -205,6 +205,32 @@ public void reset() {
             clear();
         }
     }
+    
+    /**
+     * Stop all the listeners except those writing to standard out or error. If
+     * no loggers write to standard out or error, then add one and stop the 
+     * others.
+     */
+    public void prepareToStop() {
```

Comment:
Why not simply extending from `AutoCloseable` and overriding its `close()`? I guess, there have been a discussion about this in the mailing list, though it is good to communicate the context here as well.

Reply:
The commentary in the mailing list indicates the status logger shouldn't be closed, we need to redirect the status logger to standard out iff there is no remaining listener that writes to a file.  If the log4j code is running in the container the prepareToStop functionality should stop the listener for the context that is shutting down. This code is flawed because it stops the listeners for all contexts. I'll push some changes that stop listeners on a context-by-context basis.